src/Map.cpp

```diff
@@ -174,7 +174,7 @@ bool cMap::UpdatePixel(unsigned int a_X, unsigned int a_Z)
 			}
 
 			// Multiply base color ID by 4 and add brightness ID
-			m_PixelData = ColourID * 4 + BrightnessID[Clamp<size_t>(static_cast<size_t>(Height / (ChunkHeight / BrightnessID.size())), 0, BrightnessID.size() - 1)];
+			m_PixelData = ColourID * 4 + BrightnessID[Clamp<size_t>(static_cast<size_t>(static_cast<UInt32>(Height) / (static_cast<UInt32>(ChunkHeight) / BrightnessID.size())), 0, BrightnessID.size() - 1)];
```

Comment:
`UInt32` should be used only for situations where you know exactly 32 bits are used (network protocol, disk storage etc); in other places use the platform native types, `unsigned` etc.